mne/io/base.py

```diff
@@ -33,17 +33,156 @@
+        else:
+            index = default_index
+
+        if picks is None:
+            picks = list(range(self.info['nchan']))
+        else:
+            if isinstance(self, (Epochs, EpochsArray)):
+                if not in1d(picks, np.arange(len(self.events))).all():
+                    raise ValueError('At least one picked channel is not present '
+                                     'in this epochs instance.')
+            if isinstance(self, (Evoked, EvokedArray)):
+                if not in1d(picks, np.arange(len(self.ch_names))).all():
+                    raise ValueError('At least one picked channel is not present '
+                                     'in this Evoked instance.')
+
+        if isinstance(self, (Epochs, EpochsArray)):
+            data = self.get_data()[:, picks, :]
+            times = self.times
+            n_epochs, n_picks, n_times = data.shape
+            data = np.hstack(data).T  # (time*epochs) x signals
+
+        else:
+            if isinstance(self, (RawFIFF, RawArray)):
+                data, times = self[picks, start:stop]
+            elif isinstance(self, (Evoked, EvokedArray, SourceEstimate)):
+                data = self.data[picks, :]
+                times = self.times
+            n_picks, n_times = data.shape
+            data = data.T
+
+        if copy is True:
+            data = data.copy()
+
+        times = np.round(times * scale_time)
+
+        types = [channel_type(self.info, idx) for idx in picks]
+        n_channel_types = 0
+        ch_types_used = []
+
+        scalings = _mutable_defaults(('scalings', scalings))[0]
+        for t in scalings.keys():
+            if t in types:
+                n_channel_types += 1
+                ch_types_used.append(t)
+
+        for t in ch_types_used:
+            scaling = scalings[t]
+            idx = [picks[i] for i in range(len(picks)) if types[i] == t]
+            if len(idx) > 0:
+                data[:, idx] *= scaling
+
+        # Switch for object-specific indices
+        mindex = list()
+        if isinstance(self, (RawFIFF, RawArray,
+                             Evoked, EvokedArray, SourceEstimate)):
+            mindex.append(('time', times))  
+
+        if isinstance(self, (Epochs, EpochsArray)):
+            mindex.append(('time', np.tile(times, n_epochs)))
+            id_swapped = dict((v, k) for k, v in self.event_id.items())
+            names = [id_swapped[k] for k in self.events[:, 2]]
+            mindex.append(('condition', np.repeat(names, n_times)))
+            mindex.append(('epoch', np.repeat(np.arange(n_epochs),
+                          n_times)))
+
+        assert all(len(mdx) == len(mindex[0]) for mdx in mindex)
+        col_names = [self.ch_names[k] for k in picks]
+
+        df = pd.DataFrame(data, columns=col_names)
+        [df.insert(i, k, v) for i, (k, v) in enumerate(mindex)]
+        if index is not None:
+            with warnings.catch_warnings(record=True):
```

Comment:
Please add a comment about why this is necessary. It's better to avoid warnings altogether, so it's important to justify why they are being passed silently when it's done.